pkgs/by-name/tu/turtle/package.nix

```diff
@@ -0,0 +1,93 @@
+# turtle_cli does subcommands by creating a new python subprocess through
+# sys.executable or the $PYTHON env var, when it exists. This python executable
+# must have the turtlevcs module available, hence it's pulled out into its own
+# derivation.
+{ lib
+, fetchFromGitLab
+, cacert
+, gobject-introspection
+, gtk4
+, glib
+, libadwaita
+, meld
+, python3
+, wrapGAppsHook4
+,
+}:
+let
+  pname = "turtle";
+  version = "0.6";
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "philippun1";
+    repo = pname;
+    rev = version;
+    hash = "sha256-ShZlGZeIjsMOLrdPsl6WphGZcWPrhTQHv2b7HFVm4do=";
+  };
+  turtlevcs =
+    python3.pkgs.buildPythonPackage
+      {
+        pname = "turtlevcs";
+        inherit version src;
+        format = "setuptools";
+        propagatedBuildInputs =
+          (with python3.pkgs; [
+            pygit2
+            pygobject3
+          ])
+          ++ [
+            meld
+          ];
+        doCheck = false;
+      };
+in
+python3.pkgs.buildPythonApplication {
+  inherit pname version src;
+  format = "setuptools";
+  nativeBuildInputs = [
+    wrapGAppsHook4
+    gobject-introspection
+  ];
+  buildInputs = [
+    libadwaita
+    gtk4
+    glib
+  ];
+  nativeCheckInputs = [
+    cacert
+  ];
+
+  postInstall = ''
+    # remove its own turtlevcs package, as otherwise we get duplicate modules
+    rm -r $out/lib/python*/site-packages/*
+    install -Dm644 -t $out/share/nautilus-python/extensions $src/plugins/turtle_nautilus.py
+  '';
+
+  # this runs after the wrapPythonScipts function, hence $program_PYTHONPATH is already populated.
+  # the turtle_nautilus.py plugin imports turtlevcs, hence we have to patch it
+  postFixup = ''
+    patchPythonScript "$out/share/nautilus-python/extensions/turtle_nautilus.py"
+  '';
+
+  propagatedBuildInputs = [
+    turtlevcs
+  ];
+
+  dontWrapGApps = true;
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+    # pass on python with the turtlevcs module
+    makeWrapperArgs+=( --set PYTHON ${python3.withPackages (_: [turtlevcs])}/bin/python)
```

Comment:
did you try instead of using the approach of splitting the package in two, setting `$out` as suffix or prefix to the for example `--prefix PYTHONPATH : $out/${remy-python.sitePackages}`